### PR TITLE
Fix: Change logic for _get_filename when cd header not present

### LINKED ISSUE #494: AttributeError: 'NoneType' object has no attribute 'split' when downloading from CODA
I am unsure of what is going on here. Trying to download a product, using its uuid, from https://coda.eumetsat.int.

Using the following CLI command:
`sentinelsat --uuid 79932d70-5c22-413c-9e29-493f53bc5c2d --download`

Using `sentinelsat, version 1.0.1`

I have downloaded from CODA before, so I'm a bit confused. I guess CODA is not responding with the filename in the same way anymore?

```
Found 1 products                                                                                                                                                                              
Will download 1 products using 2 workers                                                                                                                                                      
Fetching archival status: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  9.25 products/s]
79932d70-5c22-413c-9e29-493f53bc5c2d is online. Starting download                                                                                                                             
There was an error downloading S3B_OL_2_WFR____20210605T091935_20210605T092235_20210605T113051_0179_053_150_1980_MAR_O_NR_003                                                                 
Traceback (most recent call last):                                                                                                                                                            
  File "/home/ubuntu/.cache/pypoetry/virtualenvs/dhus-test-NW2MfJEz-py3.8/lib/python3.8/site-packages/sentinelsat/sentinel.py", line 890, in _download_online_retry                           
    return self.download(id, directory_path, checksum, **kwargs)                                                                                                                              
  File "/home/ubuntu/.cache/pypoetry/virtualenvs/dhus-test-NW2MfJEz-py3.8/lib/python3.8/site-packages/sentinelsat/products.py", line 178, in download                                         
    return sentinelsat.SentinelAPI.download(self, id, directory_path, checksum, **kwargs)                                                                                                     
  File "/home/ubuntu/.cache/pypoetry/virtualenvs/dhus-test-NW2MfJEz-py3.8/lib/python3.8/site-packages/sentinelsat/sentinel.py", line 511, in download                                         
    filename = self._get_filename(product_info)                                                                                                                                               
  File "/home/ubuntu/.cache/pypoetry/virtualenvs/dhus-test-NW2MfJEz-py3.8/lib/python3.8/site-packages/sentinelsat/sentinel.py", line 590, in _get_filename                                    
    filename = cd.split("=", 1)[1].strip('"')                                                                                                                                                 
AttributeError: 'NoneType' object has no attribute 'split'                                                                                                                                    
9 retries left
```

### PR DESCRIPTION
**Pull request by rbrishabh**
Fixes #494

Hey @j08lue, how does this look?
Let me know if this needs any changes! Does this need a test case? If yes, can you guide me a bit on how to add that?

Thank you :)

### CODE REVIEW

File: sentinelsat/sentinel.py [hunk 1]
```diff
@@ -574,20 +574,21 @@ def _download_outer(self, product_info: Dict[str, Any], path: Path, verify_check
...
+            req = self.session.head(product_info["url"])
             _check_scihub_response(req, test_json=False)
-            filename = req.text
-            # This should cover all currently existing file types: .SAFE, .SEN3, .nc and .EOF
-            filename = filename.replace(".SAFE", ".zip")
-            filename = filename.replace(".SEN3", ".zip")
-            return filename
-        req = self.session.head(product_info["url"])
+            cd = req.headers.get("Content-Disposition")
+            if cd:
```
Comment (j08lue) [inline sentinelsat/sentinel.py:581]:
```suggestion
            if cd is not None:
```

**Review by j08lue — COMMENTED** [review]
Nice, that is just the way to do it.

I guess we need a test for this, though. Could you add a test against CODA and record a VHR cassette?

**Comment by rbrishabh** [comment]
Hi @j08lue ! Not 100% sure how to write a test, could you please help me with that?

**Comment by j08lue** [comment]
@rbrishabh sorry for being so slow on this. I just started the CI. 

> Not 100% sure how to write a test, could you please help me with that?

I would now see which test covers the new code including the two branches you introduced (in particular `cd` not present) and then look into adding a case for your branch somehow.

**Comment by j08lue** [comment]
Seems like all is actually covered: https://app.codecov.io/gh/sentinelsat/sentinelsat/compare/495/diff. Thanks @rbrishabh, merging now.

**Pull request merged**